Feature/UI/TitlePage

AyaazStreet opened the pull request:
- Fixed any routing bugs
- Added TitlePage UI
- Added Chapter Lists
- Added Pagination to Chapter Lists

AyaazStreet committed:
fix: :bug: Fixed router Links

Imported Link from @tanstack/react-router and added them to the HomePage. Also added some margin styling.

AyaazStreet committed:
refactor: :truck: Changed API

Changed API calls baseUrl

AyaazStreet committed:
feat: :lipstick: Created UI for Manga Title page

Added UI for TitlePage and fixed any routing error bugs

AyaazStreet committed:
feat: :sparkles: Created Chapter Lists

Created Lists for chapters in TitlePage. Added Pagination with a limit of 10 chapters per page.

theadley commented on src/components/ChapterList.tsx:
```diff
@@ -0,0 +1,68 @@
+        keepPreviousData: true,
+    })
+
+    console.log(chapters)
```
👀

theadley commented on src/components/ChapterListItem.tsx:
```diff
@@ -0,0 +1,71 @@
+    })
+
+    console.log(chapterDetails)
+    console.log(chapterDetails?.chapter?.created_at?.slice(0, 10))
```
👀

theadley commented on src/components/ChapterListItem.tsx:
```diff
@@ -0,0 +1,71 @@
+                <div>
+                    <img
+                        className="object-cover object-top h-32 w-60 rounded"
+                        src={`https://meo3.comick.pictures/${chapterDetails?.chapter?.md_images?.[0].b2key}`}
```
Keep going with the optional chaining and then add a fallback

theadley commented on src/components/NewManga.tsx:
```diff
@@ -1,24 +1,30 @@
+            <h2 className="py-4 text-xl h-1/5">New Manga Chapters</h2>
+            <div className="carousel w-full px-8 space-x-2 mb-4 h-4/5">
                 {updates
                     .filter((manga) => (manga.content_rating = 'safe'))
```
This is assignment `=` and not comparison `===`

theadley commented on src/components/NewManga.tsx:
```diff
@@ -1,24 +1,30 @@
+                            >
                                 <img
                                     className="object-cover object-top w-full h-full"
                                     src={`https://meo3.comick.pictures/${manga?.md_comics?.md_covers?.[0].b2key}`}
```
As before. More chaining, safe fallback.

theadley commented on src/components/TopTenManga.tsx:
```diff
@@ -1,25 +1,31 @@
 
 function TopTenManga({ top }: { top: TopComics }) {
     const topTen = top?.recentRank
         .filter((manga) => (manga.content_rating = 'safe'))
```
Assignment not comparison

theadley commented on src/components/TopTenManga.tsx:
```diff
@@ -1,25 +1,31 @@
+                            >
                             <img
                                 className="object-cover object-top w-full h-full"
                                 src={`https://meo3.comick.pictures/${manga?.md_covers?.[0].b2key}`}
```
As above

theadley commented on src/components/TrendingManga.tsx:
```diff
@@ -18,7 +18,7 @@ function TrendingManga({
                 <img
-                    className="object-cover object-top w-full"
+                    className="object-cover object-top  w-full"
                     src={`https://meo3.comick.pictures/${trending?.md_comics?.md_covers?.[0].b2key}`}
```
As above

theadley commented on src/pages/TitlePage.tsx:
```diff
@@ -1,7 +1,112 @@
+            mangaDetails?.comic?.hid ||
+            mangaDetails?.comic?.md_comic_md_genres?.[0]?.md_genres?.name ||
+            mangaDetails?.comic?.md_covers?.[0]?.b2key ||
+            mangaDetails?.comic?.desc ||
```
I'm not certain, but I think these should be and `&&` not or `||`

theadley commented on src/pages/TitlePage.tsx:
```diff
@@ -1,7 +1,112 @@
+        queryFn: () => getMangaBySlug(manga),
+    })
+
+    console.log(mangaDetails)
```
👀

theadley commented on src/pages/TitlePage.tsx:
```diff
@@ -1,7 +1,112 @@
+                    {mangaDetails?.comic?.chapter_count} chapter
+                    {mangaDetails?.comic?.chapter_count > 1 && 's'}
+                </p>
+                <button className="btn btn-primary my-4 w-full">
```
What does this button do?

theadley reviewed (changes requested):
Minor fixes needed and console logs removed

AyaazStreet committed:
fix: :sparkles: PR Fixes

Removed console logs. Added Link on Read Now button. Added fallbacks to images and other stuff in ChapterListItem. Removed unnessory code.  Fix filter condition.

AyaazStreet committed:
refactor: :mute: Remove console logs

Oops missed some console logs.

theadley commented on src/components/NewManga.tsx:
```diff
@@ -1,32 +1,38 @@
 
 function NewManga({ updates }: { updates: Comic[] }) {
+    console.log(updates)
+    console.log(updates?.[0]?.md_comics?.md_covers?.[0]?.b2key)
```
Dude

theadley reviewed (changes requested):
Please just remove the console.logs

theadley reviewed (approved):
Yey 🎉
